chiselFrontend/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -391,39 +464,55 @@ sealed class IntervalRange(
     }
   }
 
-  // TODO: (chick) How to implement properly? (Angie)
+  /** Multiply this by that, here we return a fully unknown range,
+    * firrtl's range inference can figure this out
+    * @param that
+    * @return
+    */
   override def *(that: IntervalRange): IntervalRange = {
-    IntervalRange(firrtlir.UnknownBound, firrtlir.UnknownBound, firrtlir.UnknownWidth)
+    IntervalRange.Unknown
   }
 
-  private def doFirrtlOp(op: firrtlir.PrimOp, that: IntervalRange): IntervalRange = {
-    PrimOps.set_primop_type(
-      firrtlir.DoPrim(op,
-        Seq(firrtlir.Reference("a", this), firrtlir.Reference("b", that)), Nil,firrtlir.UnknownType)
-    ).tpe match {
-      case i: firrtlir.IntervalType => IntervalRange(i.lower, i.upper, i.point)
-      case other => sys.error("BAD!")
-    }
+  /** Add that to this, here we return a fully unknown range,
+    * firrtl's range inference can figure this out
+    * @param that
+    * @return
+    */
+  override def +&(that: IntervalRange): IntervalRange = {
+    IntervalRange.Unknown
   }
 
-  private def doFirrtlOp(op: firrtlir.PrimOp, that: Int): IntervalRange = {
```

Comment:
Why did you get rid of doFirrtlOp? This enabled Chisel reuse the FIRRTL calculations of known range inference for these types of ops, instead of defaulting to IntervalRange.Unknown. This seems like a regression in behavior, where Chisel is always kicking the can to FIRRTL instead of resolving ranges immediately when it can.

Reply:
restored both forms